internal/server/edit_my_details.go

```diff
@@ -33,6 +34,21 @@ func editMyDetails(logger *log.Logger, client EditMyDetailsClient, tmpl Template
 			return
 		}
 
+		myPermissions, err := client.MyPermissions(r.Context(), r.Cookies())
+		if err == sirius.ErrUnauthorized {
+			http.Redirect(w, r, siriusURL+"/auth", http.StatusFound)
+			return
+		} else if err != nil {
+			logger.Println("myDetails:", err)
+			http.Error(w, "Could not connect to Sirius", http.StatusInternalServerError)
+			return
+		}
+
+		if !hasPermission("user", "patch", myPermissions) {
```

Comment:
Will we ever care about having the list of permissions, or always be checking if you _have_ some permissions? Maybe it would be better to have `client.HasPermission(r.Context(), r.Cookies(), "user", "patch")` vs. this?